src/main/java/seedu/address/model/event/IsolatedEventList.java

```diff
@@ -56,6 +56,31 @@ public IsolatedEvent checkClashingIsolatedEvent(LocalDateTime startDate, LocalDa
         return null;
     }
 
+    /**
+     * Check to see if the edited event time period clashes with the existing isolated events in the list.
+     * @param event edited isolated event
+     * @param index index of edited event
+     * @throws EventConflictException to be thrown when it overlaps
+     */
+    public void checkOverlapping(IsolatedEvent event, int index) throws EventConflictException {
+        Iterator<IsolatedEvent> it = isolatedEvents.iterator();
+        LocalDateTime start = event.getStartDate();
+        LocalDateTime end = event.getEndDate();
+        IsolatedEvent currEvent;
+        int counter = 0;
+
+        while (it.hasNext()) {
+            currEvent = it.next();
+            if (counter == index) {
+                continue;
+            }
+            if (start.isBefore(currEvent.getEndDate()) && currEvent.getStartDate().isBefore(end)) {
```

Comment:
If I am not wrong your currently implementation only checks for the user changes the t/END_DATE
newEvent 11:00 - ~12:00~ 15:00
compareEvent 13:00- 16:00 --> Throws error

But how about cases like this where the user change the f/START_DATE
newEvent ~15:00~ 13:00 - 17:00
compareEvent  12:00 - 15:00